beancount/parser/printer.py

```diff
@@ -349,7 +350,11 @@ def Note(self, entry, oss):
         self.write_metadata(entry.meta, oss)
 
     def Document(self, entry, oss):
-        oss.write('{e.date} document {e.account} "{e.filename}"'.format(e=entry))
+        oss.write(
+            "{e.date} document {e.account} {filename}".format(
+                e=entry, filename=json.dumps(entry.filename, ensure_ascii=False)
+            )
+        )
```

Comment:
Why `json.dumps()`?

Reply:
for double quoted escaped string, there is no way(no easy way) to do this. `repr(s)` is single quoted

Reply:
Doing what exactly? Escaping double quotes in `filename`? I don't think `filename` is expected to contain double quotes, but anyway: `filename.replace('"', '\\"')` does it.

Reply:
I find a problem here, we can not use `\` as escaping because windows use it as filepath seprator 🤔

Reply:
good question, looks like it's not

Reply:
Why the use of `json.dump()` is still there? Why was this comment marked as resolved when it is clearly not?

Reply:
sorry， forget to revert this

Reply:
Why in hell are you changing this to escape double quotes? How the heck are you getting a filename that contains double quotes on Windows? I thought that we agreed that `"` is not a valid character in filenames on Windows. I'm getting tired of pointing out changes that have nothing to do with the stated purpose of the PR. Either you start to split up your changes in meaningful commits with commit messages explaining them, or I'll stop looking at the patches all together.

Reply:
ok, i now why `json.dumps` here at the first place. becasue we parse filename wrong and still consider `\` as escaping char, so only `json.dumps` can revert that. 😅

since it's a bug of parser itself I'll just skip related tests on win32